### DISCUSSION TITLE
Skeleton Next - Update 7 - Components and Animations

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/skeleton#2602] [Announcements]
Hey everyone, Chris here from the Skeleton crew! This post is part of an ongoing [series of updates](https://github.com/skeletonlabs/skeleton/discussions/2375) as we progress towards Skeleton v3 (aka Skeleton "Next"). As usual, we've provided a quick summary post below, as well as an in-depth video walkthrough.

Today we'll cover documentation updates, preview new components, and go in depth on component animations.

## Useful Links

- [YouTube Video Walkthrough](https://youtu.be/-ZXqlhFVCl0)

## Doc Updates

First up, the [v3 doc site](https://next.skeleton.dev/) now features a round of structural improvements. This includes deeper integration of Astro's powerful [Content Collections](https://docs.astro.build/en/guides/content-collections/). This enables MDX and Markdown content to be abstracted from the file-based routes. Think of this as a local database, where data is queried and displayed on demand. The benefit of this is simpler organization of the site content, making it easier to retrieve and display data on demand. This will likely also help support our upcoming site search indexing.

### Progress

![image](https://github.com/skeletonlabs/skeleton/assets/1509726/3a03bc65-8ace-4345-ab10-63d1e882dd4e)

Additionally, Skeleton has two new components incoming. First the new [Progress](https://next.skeleton.dev/docs/components/progress/react) component. This includes a sleek design and simple prop interface, while also introducing high quality animations, including support for custom CSS animations in indeterminate mode.

### Tabs

![Screenshot 2024-04-09 at 1 25 51 PM](https://github.com/skeletonlabs/skeleton/assets/1509726/2c605dcf-1274-47e6-9d5a-4530472625bd)

The new [Tabs](https://github.com/skeletonlabs/skeleton/pull/2583) component is coming later this week. This features a fresh new design, allows for extensive customization, while retaining the simple state management from previous versions of Skeleton.  Expect this to arrive on the doc site later this week. In the meantime, see a preview in the video walkthrough linked above.

## Component Animations

Finally, we're thrilled to announce we've finally settled on a solution for animations in Skeleton components. After much [deliberation](https://github.com/skeletonlabs/skeleton/issues/2449), we've opted to lean into the strengths of each framework.

- For Svelte, that means we'll continue using the built-in [Svelte/Transition](https://svelte.dev/docs/svelte-transition) and [Svelte/Animation](https://svelte.dev/docs/svelte-animate) features.
- For React, we'll be introducing the first peer dependency in Svelte's history - [Framer Motion](https://www.framer.com/motion/)

### Framer Motion

The React ecosystem does not currently provide a first class solution for animations, which leaves a large gap in providing high fidelity components. While there are great options such as [React Spring](https://www.react-spring.dev/) and [GSAP ](https://gsap.com/), we've opted for [Framer Motion](https://www.framer.com/motion/) due to its top-notch documentation, ease of use, and powerful feature set. It even received a shoutout in the recent [Tailwind v4 preview](https://tailwindcss.com/blog/tailwindcss-v4-alpha#css-first-configuration) (near the end). To learn more, checkout their list of [interactive examples](https://www.framer.com/motion/examples/).

## What's Next

With animations sorted, we'll continue to build new components [new components](https://github.com/skeletonlabs/skeleton/pull/2599). We'll also be putting more focus into nailing down a great solution for [automated documentation](https://github.com/skeletonlabs/skeleton/issues/2399). This enables us to parse raw code configuration settings, such as props, directly from the code and display this information directly on each feature's page. Expect more updates soon!

**Comment by withden** [comment]
Awesome work @endigo9740. I'm playing with the themes and components(next version - v3), it's really good. Glad to see good progress towards v3.
